django/db/models/sql/query.py

```diff
@@ -1077,10 +1077,15 @@ def resolve_lookup_value(self, value, can_reuse, allow_joins):
         elif isinstance(value, (list, tuple)):
             # The items of the iterable may be expressions and therefore need
             # to be resolved independently.
-            return type(value)(
+            values = (
                 self.resolve_lookup_value(sub_value, can_reuse, allow_joins)
                 for sub_value in value
             )
+            type_ = type(value)
+            if hasattr(type_, '_make'):
+                # namedtuple
+                return type_._make(values)
```

Comment:
We can use `_make` to detect a `namedtuple` but I would still use encapsulation
```
return type(value)(*values)
```

Reply:
`_make` *is* [documented as a public API](https://docs.python.org/3.8/library/collections.html#collections.somenamedtuple._make) so I think it's not breaking encapsulation to use it, but okay